app/javascript/react/screens/App/Overview/screens/MappingWizard/components/SourceClusterSelect/SourceClusterSelect.js

```diff
@@ -14,12 +14,28 @@ import {
 
 const SourceClusterSelect = ({
   form,
-  sourceClusters,
+  clusterMappings,
   selectSourceCluster,
-  selectedCluster
+  selectedCluster,
+  selectedClusterMapping
 }) => {
   const mappingObject = form.includes('Networks') ? 'networks' : 'datastores';
 
+  const sourceClustersWithAssociatedTargetClusters = clusterMappings.reduce(
+    (mappings, targetClusterWithSourceClusters) => {
+      const {
+        nodes: sourceClusters,
+        ...targetCluster
+      } = targetClusterWithSourceClusters;
+      const sourceToTargetMappings = sourceClusters.map(sourceCluster => ({
+        sourceCluster,
+        targetCluster
+      }));
+      return mappings.concat(sourceToTargetMappings);
```

Comment:
For future reference you can also use spread syntax (`...`) in arrays instead of `.concat()`. We may as well use it wherever we can if we're using it.

```
return [...mappings, ...sourceToTargetMappings];
```

I'll approve this one anyway because it's perfectly readable as is and it's more of a preference thing anyway.